ndcube/tests/test_ndcubesequence.py

```diff
@@ -0,0 +1,140 @@
+    # Assert output is as expected.
+    sliced_sequence = ndc[item]
+    assert isinstance(sliced_sequence, NDCubeSequence)
+    assert int(sliced_sequence.dimensions[0].value) == tuple_item[0].stop - tuple_item[0].start
+    print(sliced_sequence[0].dimensions, expected_cube0_dims, item, tuple_item)
+    assert (sliced_sequence[0].dimensions == expected_cube0_dims).all()
+
+
+@pytest.mark.parametrize("ndc, item",
+                         (
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[0]),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[1, 0:1]),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[2, 1]),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[3, 1, 0:2])
+                         ),
+                         indirect=("ndc",))
+def test_extract_ndcube(ndc, item):
+    cube = ndc[item]
+    tuple_item = item if isinstance(item, tuple) else (item,)
+    expected_cube_dims = derive_sliced_cube_dims(ndc.data[tuple_item[0]].dimensions, tuple_item)
+    assert isinstance(cube, NDCube)
+    assert (cube.dimensions == expected_cube_dims).all()
+
+
+@pytest.mark.parametrize("ndc, item, expected_common_axis",
+                         (
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[:, 0], 0),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[:, 0:1, 0:2], 1),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[:, :, :, 1], 1),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[:, :, 0], None)
+                         ),
+                         indirect=("ndc",))
+def test_slice_common_axis(ndc, item, expected_common_axis):
+    sliced_sequence = ndc[item]
+    assert sliced_sequence._common_axis == expected_common_axis
+
+
+@pytest.mark.parametrize("ndc, item, expected_dimensions",
+                         (
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[1:7],
+                                 (3 * u.pix, 2 * u.pix, [2., 3., 1.] * u.pix, 4 * u.pix)),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[1:7, 0],
+                                 (3 * u.pix, [2., 3., 1.] * u.pix, 4 * u.pix)),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[2:4],
+                                 (2 * u.pix, 2 * u.pix, 1 * u.pix, 4 * u.pix)),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[0:6],
+                                 (2 * u.pix, 2 * u.pix, 3 * u.pix, 4 * u.pix)),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[0:6, :, 0],
+                                 (3 * u.pix, 2 * u.pix, 4 * u.pix)),
+                             ("ndcubesequence_4c_ln_lt_l_cax1", np.s_[0::, 0, 0],
+                                 (4 * u.pix, 2 * u.pix))
+                         ),
+                         indirect=("ndc",))
+def test_index_as_cube(ndc, item, expected_dimensions):
```

Comment:
you and your expected values in parametrize, always do hard to read. Can you make the tuples one element a line or something so I actually stand a chance?